internal/scheduler/configuration/configuration.go

```diff
@@ -46,6 +46,12 @@ type Configuration struct {
 	InternedStringsCacheSize uint32 `validate:"required"`
 	// How often the scheduling cycle should run
 	CyclePeriod time.Duration `validate:"required"`
+	// How often the job scheduling should run
+	// This is expected to be a greater value than CyclePeriod as we don't need to schedule every cycle
+	// This keeps the system more responsive as other operations happen in each cycle - such as state changes
+	SchedulePeriod time.Duration `validate:"required"`
```

Comment:
not suggesting you add this now, but I think the validation framework is sophisticated to enforce the constraint that `SchedulePeriod` > `CyclePeriod`